### PR TITLE
🐛Fix PropertyPanel after collapse a pool

### LINKED ISSUE #1150: 🐛Collapsing and Expanding Pool breaks the PropertyPanel
#### Describe your issue

Topic

#### How can others reproduce the issue?

- select a pool and collapse it
- expand it
- undo both of these steps
- select an element and try to use the PropertyPanel

#### What's your setup

- BPMN-Studio Version (`4.x.x`):
- OS (`Windows/OS X/Linux` + `version`):
- Node (`node --version`):
- NPM (`npm --version`):
- Docker (`docker version --format '{{.Server.Version}}'`):

> You can remove this section if it's irrelevant to the issue
> (for example for certain questions or feature requests).

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [ ] I've checked if this issue already exists
- [ ] I've included all the information that i think is relevant
- [ ] I've added logs and/or screenshots (if applicable)
- [ ] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by alexanderkasten**
**Changes:**

1. Fix breaking PropertyPanel while collapsing pools
2. Fix Bug when selecting the 'Process' after deleting the Participant

**Issues:**

Closes #1150 
PR: #PullRequest

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/modules/design/property-panel/indextabs/general/sections/pool/pool.ts [hunk 1]
```diff
@@ -31,6 +31,12 @@ export class PoolSection implements ISection {
   }
 
   public activate(model: IPageModel): void {
+
+    const noProcessRef: boolean = model.elementInPanel.businessObject.processRef === undefined;
```
Comment (cmg-dev) [inline src/modules/design/property-panel/indextabs/general/sections/pool/pool.ts]:
```suggestion
    const noProcessReferencePresent: boolean = model.elementInPanel.businessObject.processRef === undefined;
```

File: src/modules/design/property-panel/indextabs/general/sections/pool/pool.ts [hunk 2]
```diff
@@ -31,6 +31,12 @@ export class PoolSection implements ISection {
...
+    if (noProcessRef) {
```
Comment (cmg-dev) [inline src/modules/design/property-panel/indextabs/general/sections/pool/pool.ts]:
```suggestion
    if (noProcessReferencePresent) {
```

File: src/modules/design/property-panel/indextabs/extensions/sections/basics/basics.ts [hunk 3]
```diff
@@ -39,7 +39,10 @@ export class BasicsSection implements ISection {
   }
 
   public isSuitableForElement(element: IShape): boolean {
-    if (element.businessObject === undefined) {
+    const noElementOrBusinessObject: boolean = element === undefined
```
Comment (cmg-dev) [inline src/modules/design/property-panel/indextabs/extensions/sections/basics/basics.ts]:
```suggestion
    const elementIsUndefined: boolean = element === undefined
```

File: src/modules/design/property-panel/indextabs/extensions/sections/basics/basics.ts [hunk 4]
```diff
@@ -39,7 +39,10 @@ export class BasicsSection implements ISection {
...
+                                            || element.businessObject === undefined;
+
+    if (noElementOrBusinessObject) {
```
Comment (cmg-dev) [inline src/modules/design/property-panel/indextabs/extensions/sections/basics/basics.ts]:
```suggestion
    if (elementIsUndefined) {
```

File: src/modules/design/property-panel/indextabs/extensions/sections/basics/basics.ts [hunk 5]
```diff
@@ -39,7 +39,10 @@ export class BasicsSection implements ISection {
   }
 
   public isSuitableForElement(element: IShape): boolean {
-    if (element.businessObject === undefined) {
+    const elementIsUndefined: boolean = element === undefined
+                                            || element.businessObject === undefined;
```
Comment (SteffenKn) [inline src/modules/design/property-panel/indextabs/extensions/sections/basics/basics.ts]: Indentation looks broken should look like this: 
```
    const elementIsUndefined: boolean = element === undefined
                                     || element.businessObject === undefined;
```

**Pull request merged**